src/test/java/uk/gov/register/core/TransactionMemoizationStoreImplTest.java

```diff
@@ -0,0 +1,171 @@
+        verify(internalStore, never()).put(0, 1, newHash);
+    }
+
+    @Test
+    public void putShouldNotPersistHashDirectlyToMemoizationStore() {
+        ArrayList<byte[]> addedHashes = new ArrayList<>();
+        byte[] hash = "22905e52ce72f3880bddd5564966b15b017a038d84de41c390b2aafe28fd8452".getBytes();
+        MemoizationStore internalStore = mock(MemoizationStore.class);
+        when(internalStore.get(0, 1)).thenReturn(hash);
+
+        ArgumentCaptor<byte[]> captor = ArgumentCaptor.forClass(byte[].class);
+        doAnswer(invocation -> {
+            addedHashes.add(captor.getValue());
+            return null;
+        }).when(internalStore).put(anyInt(), anyInt(), captor.capture());
+        TransactionalMemoizationStoreImpl store = createTransactionMemoizationStore(internalStore, 0);
+
+        store.put(0, 1, hash);
+
+        assertThat(addedHashes, hasSize(0));
+    }
+
+    @Test
+    public void commitEntriesShouldCommitAllHashesToUnderlyingMemoizationStore() {
+        ArrayList<byte[]> addedHashes = new ArrayList<>();
+        byte[] hash = "22905e52ce72f3880bddd5564966b15b017a038d84de41c390b2aafe28fd8452".getBytes();
+        MemoizationStore internalStore = mock(MemoizationStore.class);
+        when(internalStore.get(0, 1)).thenReturn(hash);
+
+        ArgumentCaptor<byte[]> captor = ArgumentCaptor.forClass(byte[].class);
+        doAnswer(invocation -> {
+            addedHashes.add(captor.getValue());
+            return null;
+        }).when(internalStore).put(anyInt(), anyInt(), captor.capture());
+
+        TransactionalMemoizationStoreImpl store = createTransactionMemoizationStore(internalStore, 0);
+
+        store.put(0, 1, hash);
+        store.put(1, 1, hash);
+        assertThat(addedHashes, hasSize(0));
+
+        store.commitEntries();
+        assertThat(addedHashes, hasSize(2));
+    }
+
+    @Test
+    public void rollbackEntriesShouldNotCommitAnyHashesToUnderlyingMemoizationStore() {
+        ArrayList<byte[]> addedHashes = new ArrayList<>();
+        byte[] hash = "22905e52ce72f3880bddd5564966b15b017a038d84de41c390b2aafe28fd8452".getBytes();
+        MemoizationStore internalStore = mock(MemoizationStore.class);
+        when(internalStore.get(0, 1)).thenReturn(hash);
+
+        ArgumentCaptor<byte[]> captor = ArgumentCaptor.forClass(byte[].class);
+        doAnswer(invocation -> {
+            addedHashes.add(captor.getValue());
+            return null;
+        }).when(internalStore).put(anyInt(), anyInt(), captor.capture());
+
+        TransactionalMemoizationStoreImpl store = createTransactionMemoizationStore(internalStore, 0);
+
+        store.put(0, 1, hash);
+        store.rollbackEntries();
+
+        // Commit any and all entries to prove that temporary entries have been cleared
+        store.commitEntries();
+
+        assertThat(addedHashes, hasSize(0));
+    }
+
+    @Test
+    public void setCurrentEntryCountShouldUpdateCurrentEntryCount() {
```

Comment:
this test does not test that, there would be no difference in test if I set `store.setCurrentEntryCount(5)` yet it makes difference in SUT